pyvrp/GeneticAlgorithm.py

```diff
@@ -121,23 +125,31 @@ def run(self, stop: StoppingCriterion):
         return Result(self._best, stats, iters, end)
 
     def _educate(self, individual: Individual):
-        self._ls.search(individual)
+        def _is_new_best(indiv):
+            return indiv.is_feasible() and indiv.cost() < self._best.cost()
 
-        # Only intensify feasible, new best solutions. See also the repair
-        # step below.
-        if (
-            self._params.should_intensify
-            and individual.is_feasible()
-            and individual.cost() < self._best.cost()
-        ):
-            self._ls.intensify(individual)
+        def _add_and_register(indiv):
```

Comment:
The privateness on these helpers is a bit much: they're already internal to the `_educate` scope :).